### PR TITLE
Refactor FXIOS-7699 [v122] Start migration to using UIButton.Configuration for buttons styling

### PR DESCRIPTION
**Pull request by tonell-m**
## :scroll: Tickets

This PR is linked to the following ticket and issue but **it is just preparatory work, it does not resolve them yet**: 

- [FXIOS-7699](https://mozilla-hub.atlassian.net/browse/FXIOS-7699): https://github.com/mozilla-mobile/firefox-ios/issues/17179

## :bulb: Description

This PR is the first step to using `UIButton.Configuration` for managing button styling since using content, title and image edge insets is now deprecated. Here we introduce a new `ResizableButton` class that use this new configuration paradigm, and the old class has been renamed `LegacyResizableButton`. The next steps will be to gradually migrate all usages of this base class to use the new class one at a time.

This PR follows the dicussion here: #17558 

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by lmarceau — COMMENTED** [review]
Great first PR so we achieve this step by step 🥇

File: BrowserKit/Sources/ComponentLibrary/Buttons/LegacyResizableButton.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+/// Deprecated base button class that use content, image and title edge insets.
+/// Should be replaced gradually with ``ResizableButton``, using UIButton.Configuration instead.
+@available(*, deprecated, renamed: "ResizableButton",
+            message: "Should be replaced with ResizableButton, using UIButton.Configuration")
```
[the remark this answers is not in the document]
Comment (tonell-m) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LegacyResizableButton.swift:10]: After running a build locally I get 77 warnings, this is the current threshold according the configuration file you've linked how much headroom do you think is reasonable?

[the remark this answers is not in the document]
Comment (lmarceau) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LegacyResizableButton.swift:10]: I guess you can change the threshold to 79, so headroom of 2 warnings and let's adjust if needed. We'll be able to lower anyway once the other PRs are in 👍 So it's only temporary

[the remark this answers is not in the document]
Comment (tonell-m) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LegacyResizableButton.swift:10]: Done! 🙂

**Pull request merged**